includes/controllers/class-message.php

```diff
@@ -67,6 +74,95 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Fetches messages.
+	 *
+	 * @return WP_Rest_Response
+	 */
+	public function fetch_messages() {
+
+		// Check authentication.
+		if ( ! is_user_logged_in() ) {
+			return hp\rest_error( 401 );
+		}
+
+		// Get thread.
+		$thread_ids = hivepress()->request->get_context( 'message_thread_ids' );
+
+		// Check thread.
+		if ( ! $thread_ids ) {
+			return hp\rest_error( 400 );
+		}
+
+		// Get thread messages.
+		$thread_messages = Models\Message::query()->filter(
```

Comment:
Why fetching messages for all the threads, only the current thread seems to be relevant? For example the query may fetch unread messages from the sender of the currently opened thread

Reply:
Exactly, so here I'm getting sender to get messages from him